### ISSUE TITLE
[Feature] Allow Just Checking if ENV Var is Set - `Deno.env.has(foo)`

### ISSUE THREAD

**Issue by sno2** [denoland/deno#10349]
If you look at the `Deno.env` API, then you will notice that its API has most of the mutation/getter functions as [JavaScript Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) (both have `.get(foo)`, `.set(foo, bar)`, `.delete(foo)`).  However, the `Deno.env` object does not have a `.has(foo)` like a JS Map.

I think that it would be great if we could add a `Deno.env.has` method which takes in a single parameter being the name of the env variable to get.  The method would just check if the env variable is present (has any value).  This could easily be added in TS (using mocks for env values) with the following code:

```ts
class DenoENV {
  protected static mockEnv: Record<string, string | undefined> = {
    NO_COLOR: "",
    PATH: "",
  };

  static set(name: string, value: string) {
    DenoENV.mockEnv[name] = value;
  }

  static get(name: string): string | undefined {
    return (DenoENV.mockEnv as Record<string, string | undefined>)[name];
  }
  
  static has(name: string): boolean {
    return DenoENV.get(name) !== undefined;
  }
}

class Deno {
  static readonly env = DenoENV;
}
```

And you would use this new method like in the following example:

```ts
// yeah, I know we have `Deno.noColor` but this is still an example nevertheless
Deno.env.has("NO_COLOR"); // `true`

// this would *not* throw an error
Deno.env.has(""); // `false`

Deno.env.has("HELLO"); // `false`
Deno.env.set("HELLO", "WORLD");
Deno.env.has("HELLO"); // `true`
```

You can checkout a playground and test this mocking implementation at [this TypeScript playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAEAiCmA7A9gUQHIDVoG8BQ00ADgE7IAu8wlAJtBOWOQJbDQC2ywA1qogG4AuaACUqyEjQA8DEs0QBzADT1ycxdAA+0AK6Ia8AGbz4NAHzQAvLgKFo6APIB9AMIOAMg5HCARD6W2hAAKAIIAKgASvv62AL4A3Hi2DEys9PDkABSIYOzwwrLyytD8YCA6+arqCgCUNnZwSGhYAHScPHz8ANo5eQC6ViVlFYmEsUmEKSxsChnZuZWFijUFakVauvpGJnT4DSQZOiSI0JkIKBiYbVy8AtCQouKSMmuKKksKG3oGxoimZjUegs+qNoONCMlGNNoAALSDzPKraoraAAI2QyBA8DAJz2dgO5COJ3OzSusyyvXgdQAhJZrN9tn8aKDxuM8KBIDASfVVKk2AcwDRkIgQABPaBIfiDEmXRJsgD08ugouxMJUAEloNwUAB3aA6+CwsD8Q0AAxJLRQLkxElNaJ05Gg5BhzBgrtUzBAIHuJ3gAA9ckQsdA-iaSM74FioHgLZKWnCIJkfI5XB4vD4avFoIroKa1BVTUkc873TrkDoQHQAFQochVp0wsh6nESkhkEgxpotOMJpMZrM502GMoQeCFzsobsCePwnwRVDuTz97NKocjscT5BT-gtUdZOcLpcqHwAdS87lg-c325niYPi4cy8H+bHQA).

**Comment by sno2** [comment]
Check out #10267 in which nayeemrmn stated the following:

> The API of Deno.env reflects a map already. The more relevant point is that the "valid" keys are determined by the external environment, not some set of names statically defined by the surrounding code.

This is actually invalid as `Deno.env` doesn't have the `Deno.env.has` and a couple of the other methods listed on [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map).

**Comment by timreichen** [comment]
I like the idea of having `Deno.env.has()` next to `Deno.env.set()` and `Deno.env.get()`. Also for a more map-like syntax we should also consider being able to do `Object.fromEntries(Deno.env)` instead of `Deno.env.toObject()`. `toObject` is never defined in any web API as far as I know.

**Comment by CrimsonCodes0** [comment]
> I like the idea of having `Deno.env.has()` next to `Deno.env.set()` and `Deno.env.get()`. Also for a more map-like syntax we should also consider being able to do `Object.fromEntries(Deno.env)` instead of `Deno.env.toObject()`. `toObject` is never defined in any web API as far as I know.

(One could argue that, because it isn't a web API, it can't directly be compared to web APIs.)

`Object.fromEntries` accepts an iterable object, so if an iterator were added to the object (`[Symbol.iterator](): Iterator<[ string, string | undefined ]>`), then one could pass it direct into `Object.fromEntries`, and the `.toObject()` method could effectively be deprecated. Adding the iterator would also give `Deno.env` a more `Map`-like API too.

**Comment by timreichen** [comment]
> (One could argue that, because it isn't a web API, it can't directly be compared to web APIs.) 

Correct, but I also would add that deno should borrow existing design choises from web apis if appropriate of course. That seems to be the case here.
+1 for iterator

**Comment by jsejcksn** [comment]
I think the iterator is a good idea also, but I'm confused about this suggestion:

> Iterator<[ string, string | undefined ]>

Shouldn't the value simply be `string`? As in:
```ts
Iterator<[string, string]>
```

`string | undefined` makes sense for the getter since it's not statically known whether the variable is defined or not, but when would a defined environment variable have `undefined` as a value?

**Comment by CrimsonCodes0** [comment]
> I think the iterator is a good idea also, but I'm confused about this suggestion:
> 
> > Iterator<[ string, string | undefined ]>
> 
> Shouldn't the value simply be `string`? As in:
> ```ts
> Iterator<[string, string]>
> ```
> 
> `string | undefined` makes sense for the getter since it's not statically known whether the variable is defined or not, but when would a defined environment variable have `undefined` as a value?

Yup! Thanks for catching that, I'll update my comment appropriately.

**Comment by timreichen** [comment]
Any updates on that?
